app/lib/actions/katello/repository/index_package_groups.rb

```diff
@@ -0,0 +1,19 @@
+module Actions
+  module Katello
+    module Repository
+      class IndexPackageGroups < Actions::EntryAction
+        def plan(repository)
+          plan_self(:user_id => ::User.current.id, :id => repository.id)
+        end
+
+        def run
+          ::User.current = ::User.find(input[:user_id])
+          repo = ::Katello::Repository.find(input[:id])
+          repo.index_db_package_groups
+        ensure
+          ::User.current = nil
+        end
```

Comment:
You can actually simplify this by using the keep_current_user middlware:

for example: https://github.com/Katello/katello/blob/5794880c764a1267e484737c39516c6400fb7288/app/lib/actions/katello/repository/sync.rb#L6

most of our actions don't use this but should